src/main/java/htsjdk/samtools/sra/SRAAlignmentIterator.java

```diff
@@ -149,16 +153,17 @@ private void nextReference() {
         }
 
         try {
+            alignedIterator = null;
+            hasMoreAlignments = false;
+
             hasMoreReferences = referencesChunksIterator.hasNext();
```

Comment:
This is just a way `nextReference()` is implemented. It tries to advance to the next reference and sets `hasMoreReferences` member variable indicating if it reached to the last reference in the list and whether future call to `nextReference()` is permitted or not.
That is somewhat similar to implementation of iterators in ngs-java